instrumentation/github.com/aws/aws-sdk-go-v2/otelaws/config.go

```diff
@@ -65,3 +66,10 @@ func WithAttributeSetter(attributesetters ...AttributeSetter) Option {
 		cfg.AttributeSetter = append(cfg.AttributeSetter, attributesetters...)
 	})
 }
+
+// WithAttributeSettersConfig specifies a config used by the attribute setter functions.
+func WithAttributeSettersConfig(settersConfig *AttributeSettersConfig) Option {
```

Comment:
I think this option is way too generic. The whole purpose of functional attributes is to provide explicit options. So providing one with a config structs defeats the purpose (and can be quite confusing).

We should have an option such as `WithSNSPhoneNumber()` which enables the phone number.

Reply:
I left the config as is, just changed the option and gave the field a more specialized name.

Reply:
What's the reason for keeping a dedicated struct which holds a single attribute? We could be passing the original config struct.

Reply:
I thought about separating concerns and only passing what's actually related to the setters, but now that I look at the other packages I don't even know why in this one the config record is not present as a field in the otelMiddlewares struct. I'm thinking about simply replacing
```
type otelMiddlewares struct {
	tracer                 trace.Tracer
	propagator             propagation.TextMapPropagator
	attributeSetter        []AttributeSetter
	attributeSettersConfig AttributeSettersConfig
}
```
with
```
type otelMiddlewares struct {
	tracer trace.Tracer
	config Config
}
```
and also passing this `Config` record to the setters